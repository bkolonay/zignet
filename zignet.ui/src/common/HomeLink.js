import React, { Component } from 'react';

class HomeLink extends Component {

  render() {
    const link = this.props.grouped ? 
      <a href="/?group=true">ZigNet</a> :
      <a href="/">ZigNet</a>;

    return (
      link
    );
  }
}

export default HomeLink;
