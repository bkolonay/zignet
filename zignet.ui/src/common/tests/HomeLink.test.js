import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import HomeLink from '../HomeLink';

it('renders when not grouped', () => {
  const component = <HomeLink grouped={false} />;
  ReactDOM.render(component,document.createElement('div'));
});

it('renders when not grouped', () => {
  const component = <HomeLink grouped={true} />;
  ReactDOM.render(component,document.createElement('div'));
});

it('snapshot when not grouped', () => {
  const component = <HomeLink grouped={false} />;
  const tree = renderer.create(component).toJSON();
  expect(tree).toMatchSnapshot();
});

it('snapshot when grouped', () => {
  const component = <HomeLink grouped={true} />;
  const tree = renderer.create(component).toJSON();
  expect(tree).toMatchSnapshot();
});