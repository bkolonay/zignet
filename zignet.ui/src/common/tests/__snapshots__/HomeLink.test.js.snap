// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot when grouped 1`] = `
<a
  href="/?group=true"
>
  ZigNet
</a>
`;

exports[`snapshot when not grouped 1`] = `
<a
  href="/"
>
  ZigNet
</a>
`;
